src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/AttendanceService.kt

```diff
@@ -141,10 +176,10 @@ class AttendanceService(
       courseSchedule.courseActivity,
       offenderBooking,
     )
-      ?: throw NotFoundException("Offender program profile for offender booking with id=$bookingId and course activity id=${courseSchedule.courseActivity.courseActivityId} not found")
+      ?: throw BadDataException("Offender program profile for offender booking with id=$bookingId and course activity id=${courseSchedule.courseActivity.courseActivityId} not found")
```

Comment:
Not founds weren't appropriate and could hide genuine not found on get attendance.